### PR TITLE
Validate IP Address

### PR DESCRIPTION
**Pull request by tommotaylor**

### CODE REVIEW

File: spec/use_cases/administrator/check_if_valid_ip_spec.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      let(:address) { '192.168.1.15/24' }
+
+      it 'returns true' do
+        result = subject.execute(address)
+        expect(result).to eq(success: true)
+      end
+    end
+
+    context 'with a valid IPv6 address' do
+      let(:address) { '2001:db8:0:1234:0:567:8:1' }
```
Comment (kjdchapman) [inline spec/use_cases/administrator/check_if_valid_ip_spec.rb]: Sadly, RADIUS can't accept IPv6 addresses, so this actually needs to fail!

Expect `IPAddr` has some easy config way to do this.

File: spec/use_cases/administrator/check_if_valid_ip_spec.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      let(:address) { '172.16.254.1' }
+
+      it 'returns true' do
+        result = subject.execute(address)
+        expect(result).to eq(success: true)
+      end
+    end
+
+    context 'with a valid suffixed IPv4 address' do
+      let(:address) { '192.168.1.15/24' }
```
Comment (kjdchapman) [inline spec/use_cases/administrator/check_if_valid_ip_spec.rb]: Same as IPv6, RADIUS can't accept IP ranges.

In the future it'd be nice to have stuff like, and then the app responsible for sending to RADIUS sorts things out, but feels like we should just flat disallow everything but basic IPs for the mo.

File: spec/use_cases/administrator/check_if_valid_ip_spec.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
+describe CheckIfValidIp do
+  subject { described_class.new }
+
+  context 'invalid' do
```
Comment (kjdchapman) [inline spec/use_cases/administrator/check_if_valid_ip_spec.rb]: Might be good to test a typical "mistyped and invalid IP" here - I'd expect IPAddr to cover it, but it'll make me sleep easier at night.

`let(:address) { '192.618.0.1' }`

Comment (tommotaylor) [reply spec/use_cases/administrator/check_if_valid_ip_spec.rb]: Done

File: spec/use_cases/administrator/check_if_valid_ip_spec.rb (hunk 1 above)
Comment (tommotaylor) [reply spec/use_cases/administrator/check_if_valid_ip_spec.rb]: Great catch

File: lib/use_cases/administrator/check_if_valid_ip.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+class CheckIfValidIp
+  def execute(address)
+    { success: address_valid_for_radius?(address) }
+  end
+
+private
+
+  def address_valid_for_radius?(address)
+    return false if address.nil?
```
Comment (emileswarts) [inline lib/use_cases/administrator/check_if_valid_ip.rb]: Was wondering if this line is required?
Same with the one below actually, is a subnet a valid ipv4?

**Pull request merged**